### ISSUE TITLE
Expose easy way to compare file to text string

### ISSUE THREAD

**Issue by hadley** [r-lib/io#1]
Useful for only printing message/changing modification date if file has actually changed.

**Comment by gaborcsardi** [comment]
@hadley wouldn't you want to just hash the file / string in this case?

**Comment by hadley** [comment]
Yeah, exactly - but handling that 100% correctly with line endings when you have (e.g.) a vector of lines, is a little tricky
